src/libraries/Directory.Build.targets

```diff
@@ -243,7 +243,7 @@
 
   <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->
   <Choose>
-    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
+    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(SupportedOSPlatforms)' != '' or '$(UnsupportedOSPlatforms)' != '') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

Comment:
very good idea to not always require `IncludePlatformAttributes` when `SupportedOSPlatforms` or `UnsupportedOSPlatforms` are used 👍 

You could simplify the assemblies annotated in https://github.com/dotnet/runtime/pull/40612 and replace

```xml
    <IncludePlatformAttributes>true</IncludePlatformAttributes>
    <UnsupportedOSPlatforms>browser</UnsupportedOSPlatforms>
```

with just

```xml
    <UnsupportedOSPlatforms>browser</UnsupportedOSPlatforms>
```